www/install/step_upgrade/step3.php

```diff
@@ -43,55 +43,59 @@
 require_once '../steps/functions.php';
 $template = getTemplate('templates');
 
+require_once realpath(dirname(__FILE__) . "/../../../config/centreon.config.php");
+include_once _CENTREON_PATH_ . "/www/class/centreonDB.class.php";
+
 $title = _('Release notes');
 
-if (is_file('../RELEASENOTES.html')) {
-    $contents = "<div id='releasenotes'></div>";
-} else {
-    $releasenotesContent = '';
-    if (file_exists('../RELEASENOTES')) {
-        $releasenotesContent = file_get_contents('../RELEASENOTES');
+$db = new CentreonDB();
+$res = $db->query("SELECT `value` FROM `informations` WHERE `key` = 'version'");
+$row = $res->fetchRow();
+$current = $row['value'];
+
+$_SESSION['CURRENT_VERSION'] = $current;
+
+/*
+** To find the final version that we should update to, we will look in
+** the www/install/php directory where all PHP update scripts are
+** stored. We will extract the target version from the filename and find
+** the farthest version to the current version.
+*/
+$next = '';
+if ($handle = opendir('../php')) {
+    while (false !== ($file = readdir($handle))) {
+        if (preg_match('/Update-([a-zA-Z0-9\-\.]+)\.php/', $file, $matches)) {
+            if ((version_compare($current, $matches[1]) < 0) &&
+                (empty($next) || (version_compare($next, $matches[1]) < 0))) {
+                $next = $matches[1];
+            }
+        }
     }
-    $contents = "<textarea cols='100' rows='30' readonly>" . $releasenotesContent . "</textarea>";
+    closedir($handle);
 }
 
+$majors = explode('.', $next);
```

Comment:
prefer usage of preg_match : `preg_match('/^(\d+\.\d+)/', $next, $matches);`
Then, $matches[1] will be the major version (eg: 19.04)

Reply:
done